internal/services/token-provider.go

```diff
@@ -65,7 +65,7 @@ func (issuer *TokenIssuer) GenerateExtraToken(username string, email string, has
 	return &signedToken, err
 }
 
-func (issuer *TokenIssuer) GenerateUserToken(groups []string, username string, email string, hasAdminAccess bool, hasApplicationAccess bool, hasOpsAccess bool) (*string, error) {
+func (issuer *TokenIssuer) GenerateUserToken(groups []string, username string, email string, hasAdminAccess bool, hasApplicationAccess bool, hasOpsAccess bool, hasViewerAccess bool) (*string, error) {
```

Comment:
I feel this is never gonna end with features, can you think of an alternative way to be more generic?